test/proposal.test.ts

```diff
@@ -147,28 +147,27 @@ describe("proposal", function () {
         hre.ethers,
       );
 
-      let virtualTokenCreationTxResponse: TransactionResponse;
-
-      for (const step of deploymentStepsIntoArray(steps)) {
-        if (step == steps.virtualCowTokenCreationTransaction) {
-          virtualTokenCreationTxResponse = await execSafeTransaction(
-            gnosisDao,
-            step,
-            [gnosisDaoOwner],
-          );
-        } else {
-          await execSafeTransaction(gnosisDao, step, [gnosisDaoOwner]);
-        }
+      const allCreatedProxies = [];
+      for (const step of groupWithMultisendCallOnly(
+        deploymentStepsIntoArray(steps),
+        gnosisSafeManager.getDeploymentAddresses().multisendCallOnly,
+      )) {
+        const response = await execSafeTransaction(gnosisDao, step, [
+          gnosisDaoOwner,
+        ]);
+        allCreatedProxies.push(
+          ...(await contractsCreatedWithCreateCall(
+            response,
+            gnosisSafeManager.createCall.address,
+          )),
+        );
       }
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      expect(virtualTokenCreationTxResponse!).not.to.be.undefined;
 
-      const proxies = await contractsCreatedWithCreateCall(
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        virtualTokenCreationTxResponse!,
-        gnosisSafeManager.createCall.address,
+      const proxiesMinusKnownAddresses = allCreatedProxies.filter(
```

Comment:
Oh no, now you needed anyways to come up with this logic. Fortunately, it is not complex at all.